app/src/org/commcare/dalvik/activities/FormAndDataSyncer.java

```diff
@@ -128,30 +129,63 @@ public void syncData(boolean formsToSend,
             @Override
             protected void deliverResult(CommCareHomeActivity receiver, Integer result) {
                 receiver.getUIController().refreshView();
+                String reportSyncLabel;
+                int reportSyncValue;
 
                 //TODO: SHARES _A LOT_ with login activity. Unify into service
                 switch (result) {
                     case DataPullTask.AUTH_FAILED:
+                        reportSyncLabel = GoogleAnalyticsFields.LABEL_SYNC_FAILURE;
```

Comment:
hm, this looks pretty tough to manage to me. I wonder whether it's time to move these integer return codes to be enums that have tagged attributes that we can return such that we can just issue the results to the analytics framework with one call, rather than one-per-case.